base/docs/basedocs.jl

```diff
@@ -2529,9 +2529,14 @@ true
 julia> "Hello!" isa IntOrString
 true
 
-julia> 1.0 isa IntOrString
+julia> 1.0 isa IntOrString # Float is neither Int, nor AbstractString
 false
+
+julia> IntOrString(1) # error, cannot instantiate a Union type
```

Comment:
What thing about it look confusing? Honest question. Standing my point above, coming from C/C++, this is what I would think is the way of getting a `IntOrString`.

Reply:
Mostly, it's because in julia, this looks like it could be a conversion, not necessarily a constructor. Note that it is also not possible to instantiate a `Number`, but this works just fine:
```julia-repl
julia> Number(1)
1
```

Because of this, some Unions you try this with won't throw an error:
```julia-repl
julia> Union{BigInt, Real}(1.0)
1.0
```
This one works (I'm pretty sure) because julia tries to resolve which of the two conversions/constructors are "more applicable" to the number `1.0`, and finds that the `Real` case is better, so it becomes `Real(1.0)`.

The only important property inherent to Unions (arguably) is that `T <: Union{T, S} && S <: Union{T, S}`

The fact that they can't be instantiated is true, but I don't think it belongs here. Mentioning it explicitly makes it seem uniquely important to Unions, which I think is why I consider it confusing.

Reply:
The issue with this is 

```julia
julia> Union{BigInt, Real}
Real
```

so it's not really a `Union` after all. Something similar actually does highlight your point:

```julia
julia> Union{BigInt, Int}
Union{Int64, BigInt}

julia> Union{BigInt, Int}(2)
2
```

This works because of the fallback

```julia
(::Type{T})(x::T) where {T<:Number} = x
```
which seems to blur the lines between conversion and construction (is this a constructor for concrete types and a conversion for abstract ones?).

Reply:
Well, every conversion is in itself a construction. While I think about it now, it might be better to take it out as this is not something dependent only on the Type, but methods defined on it.